lib/Loop/Loop.php

```diff
@@ -88,7 +88,7 @@ public function setInterval(callable $cb, float $timeout): array
     /**
      * Stops a running interval.
      *
-     * @param array<int, mixed> $intervalId
+     * @param list<mixed> $intervalId
      */
     public function clearInterval(array $intervalId): void
```

Comment:
the implementation of this method looks weird. do we have a unit test for it? 

I don't think it can work like that, since we just get a array-by-value and not by reference?

Reply:
The return of `setInterval` is:
```
        // Really the only thing that matters is returning the $keepGoing
        // boolean value.
        //
        // We need to pack it in an array to allow returning by reference.
        // Because I'm worried people will be confused by using a boolean as a
        // sort of identifier, I added an extra string.
        return ['I\'m an implementation detail', &$keepGoing];
```

I updated the param doc everywhere that is passed around to be more specific.

Reply:
The code of this function is strange:
```
 $intervalId[1] = false;
```

There is a test case that uses `clearInterval` to stop running a function after 5 or 6 intervals. The test works. IMO the code works because the thin at array index 1 is actually the address of the boolean. The array passed to `clearInterval` is passed normally - the reference is embedded in the array.

Reply:
I get it now, thanks for checking.